example/src/main/java/org/mozilla/fxa_data/example/FirefoxDataInRecyclerViewExampleActivity.kt

```diff
@@ -0,0 +1,233 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fxa_data.example
+
+import android.content.Intent
+import android.os.Bundle
+import android.support.annotation.StringRes
+import android.support.v7.app.AppCompatActivity
+import android.support.v7.widget.DividerItemDecoration
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.RecyclerView
+import android.view.View
+import android.widget.Button
+import android.widget.TextView
+import org.mozilla.fxa_data.FirefoxData
+import org.mozilla.fxa_data.FirefoxDataException
+import org.mozilla.fxa_data.login.FirefoxDataLoginManager
+import org.mozilla.fxa_data.download.FirefoxDataClient
+import org.mozilla.fxa_data.download.HistoryRecord
+import org.mozilla.fxa_data.download.FirefoxDataResult
+import java.lang.Exception
+
+import java.lang.ref.WeakReference
+import java.util.Arrays
+import java.util.Collections
+
+/**
+ * An example Activity that which displays a user's history in a [RecyclerView] after they log in.
+ * If they're already logged in, the data will just be shown. Once the data is shown, the user can click
+ * "Sign out" to sign out.
+
+ * This Activity will switch between four states:
+ * - Prompt for sign in
+ * - Loading
+ * - Error logging in or fetching Firefox data
+ * - Display user's history.
+
+ * This is a fairly realistic use case. For a simpler example, see [SimpleExampleActivity].
+ */
+public class FirefoxDataInRecyclerViewExampleActivity : AppCompatActivity() {
+
+    private enum class UIState {
+        SIGN_IN_PROMPT, LOADING, ERROR, SHOW_DATA
+    }
+
+    private lateinit var loginManager: FirefoxDataLoginManager
+    private lateinit var historyAdapter: FirefoxHistoryAdapter
+
+    private var isWaitingForCallback = false
+
+    private lateinit var containerViews: List<View>
+
+    private lateinit var fxDataContainer: View
+    private lateinit var fxDataView: RecyclerView
+    private lateinit var fxDataSignOutButton: Button
+
+    private lateinit var loadingContainer: View
+
+    // For sign in & error states.
+    private lateinit var explanationContainer: View
+    private lateinit var explanationView: TextView
+    private lateinit var explanationButton: Button
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_firefox_data_in_recycler_view_example)
+
+        supportActionBar?.title = FirefoxDataInRecyclerViewExampleActivity::class.simpleName
+
+
+        findViewsById()
```

Comment:
You can use the kotlin android extensions to get rid of this method and the variables:

http://kotlinlang.org/docs/tutorials/android-plugin.html